oxide-vpc/src/engine/gateway/arp.rs

```diff
@@ -82,7 +85,8 @@ pub fn setup(
                 cfg.guest_mac,
             )));
             let rule = Rule::new(1, action);
-            layer.add_rule(Direction::In, rule.finalize());
+            // Incoming ARP requests will hit the "firewall" layer first so add the rule there.
+            pb.add_rule("firewall", Direction::In, rule.finalize())?;
```

Comment:
These changes should not be here. The firewall layer should only allow what control plane tells it to allow. Instead, for testing, you can use the `opteadm add-fw-rule` command to add an inbound rule to allow all traffic, which includes ARP.

```
$ opteadm add-fw-rule --action allow --dir in --hosts any --ports any --priority 10 --protocol any -p <port name>
```

It's also worth noting that the "external IP hack" is going away ASAP.

Reply:
Ah, I wanted to test other firewall rules worked as expected (e.g. SSH/ICMP-in) and I couldn't see a way to add more fine-grained rules to match just ARP. Since it is just a hack I'll remove that commit